Incompatible kafka version for kafka connector and cdc-debezium connector

When both `hazelcast-jet-kafka` and `hazelcast-jet-cdc-debezium` are on Jet server classpath then there can occur exception related to incompatible kafka version. It started to happen from this commit https://github.com/hazelcast/hazelcast-jet/commit/264469f070492b19b01da50be9cfd8ecfd78f94e

I expect it is due to `org.apache.kafka:kafka-clients:jar:2.2.0:compile` in `hazelcast-jet-kafka` and `org.apache.kafka:kafka-clients:jar:2.3.1:compile` in `hazelcast-jet-cdc-debezium` (run `mvn dependency:tree` to see it).

For example following exception is thrown for our `StatefulMapTest` soak test (see steps for reproduce to see how to run it):
```
com.hazelcast.jet.JetException: Exception in ProcessorTasklet{StatefulMapTest/writeKafka(StatefulMapTesttxStatefulMapTest)#0}: java.lang.RuntimeException: Incompatible KafkaProducer version
	at com.hazelcast.jet.impl.execution.TaskletExecutionService$BlockingWorker.run(TaskletExecutionService.java:304) ~[hazelcast-jet-4.2-SNAPSHOT.jar:4.2-SNAPSHOT]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) ~[?:1.8.0_181]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) ~[?:1.8.0_181]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) ~[?:1.8.0_181]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) ~[?:1.8.0_181]
	at java.lang.Thread.run(Thread.java:748) ~[?:1.8.0_181]
Caused by: java.lang.RuntimeException: Incompatible KafkaProducer version
	at com.hazelcast.jet.kafka.impl.ResumeTransactionUtil.getValue(ResumeTransactionUtil.java:122) ~[hazelcast-jet-kafka-4.2-SNAPSHOT.jar:?]
	at com.hazelcast.jet.kafka.impl.ResumeTransactionUtil.getValue(ResumeTransactionUtil.java:113) ~[hazelcast-jet-kafka-4.2-SNAPSHOT.jar:?]
	at com.hazelcast.jet.kafka.impl.ResumeTransactionUtil.resumeTransaction(ResumeTransactionUtil.java:48) ~[hazelcast-jet-kafka-4.2-SNAPSHOT.jar:?]
	at com.hazelcast.jet.kafka.impl.WriteKafkaP.recoverTransaction(WriteKafkaP.java:184) ~[hazelcast-jet-kafka-4.2-SNAPSHOT.jar:?]
	at com.hazelcast.jet.kafka.impl.WriteKafkaP.lambda$init$89868699$1(WriteKafkaP.java:102) ~[hazelcast-jet-kafka-4.2-SNAPSHOT.jar:?]
	at com.hazelcast.function.ConsumerEx.accept(ConsumerEx.java:46) ~[hazelcast-jet-4.2-SNAPSHOT.jar:4.2-SNAPSHOT]
	at com.hazelcast.jet.impl.processor.TransactionPoolSnapshotUtility.restoreFromSnapshot(TransactionPoolSnapshotUtility.java:241) ~[hazelcast-jet-4.2-SNAPSHOT.jar:4.2-SNAPSHOT]
	at com.hazelcast.jet.impl.processor.TwoPhaseSnapshotCommitUtility.restoreFromSnapshot(TwoPhaseSnapshotCommitUtility.java:178) ~[hazelcast-jet-4.2-SNAPSHOT.jar:4.2-SNAPSHOT]
	at com.hazelcast.jet.kafka.impl.WriteKafkaP.restoreFromSnapshot(WriteKafkaP.java:171) ~[hazelcast-jet-kafka-4.2-SNAPSHOT.jar:?]
	at com.hazelcast.jet.impl.processor.ProcessorWrapper.restoreFromSnapshot(ProcessorWrapper.java:133) ~[hazelcast-jet-4.2-SNAPSHOT.jar:4.2-SNAPSHOT]
	at com.hazelcast.jet.impl.execution.ProcessorTasklet.processInbox(ProcessorTasklet.java:412) ~[hazelcast-jet-4.2-SNAPSHOT.jar:4.2-SNAPSHOT]
	at com.hazelcast.jet.impl.execution.ProcessorTasklet.stateMachineStep(ProcessorTasklet.java:289) ~[hazelcast-jet-4.2-SNAPSHOT.jar:4.2-SNAPSHOT]
	at com.hazelcast.jet.impl.execution.ProcessorTasklet.stateMachineStep(ProcessorTasklet.java:284) ~[hazelcast-jet-4.2-SNAPSHOT.jar:4.2-SNAPSHOT]
	at com.hazelcast.jet.impl.execution.ProcessorTasklet.stateMachineStep(ProcessorTasklet.java:266) ~[hazelcast-jet-4.2-SNAPSHOT.jar:4.2-SNAPSHOT]
	at com.hazelcast.jet.impl.execution.ProcessorTasklet.call(ProcessorTasklet.java:240) ~[hazelcast-jet-4.2-SNAPSHOT.jar:4.2-SNAPSHOT]
	at com.hazelcast.jet.impl.execution.TaskletExecutionService$BlockingWorker.run(TaskletExecutionService.java:293) ~[hazelcast-jet-4.2-SNAPSHOT.jar:4.2-SNAPSHOT]
	... 5 more
Caused by: java.lang.NoSuchFieldException: nextSequence
	at java.lang.Class.getDeclaredField(Class.java:2070) ~[?:1.8.0_181]
	at com.hazelcast.jet.kafka.impl.ResumeTransactionUtil.getValue(ResumeTransactionUtil.java:118) ~[hazelcast-jet-kafka-4.2-SNAPSHOT.jar:?]
	at com.hazelcast.jet.kafka.impl.ResumeTransactionUtil.getValue(ResumeTransactionUtil.java:113) ~[hazelcast-jet-kafka-4.2-SNAPSHOT.jar:?]
	at com.hazelcast.jet.kafka.impl.ResumeTransactionUtil.resumeTransaction(ResumeTransactionUtil.java:48) ~[hazelcast-jet-kafka-4.2-SNAPSHOT.jar:?]
	at com.hazelcast.jet.kafka.impl.WriteKafkaP.recoverTransaction(WriteKafkaP.java:184) ~[hazelcast-jet-kafka-4.2-SNAPSHOT.jar:?]
	at com.hazelcast.jet.kafka.impl.WriteKafkaP.lambda$init$89868699$1(WriteKafkaP.java:102) ~[hazelcast-jet-kafka-4.2-SNAPSHOT.jar:?]
	at com.hazelcast.function.ConsumerEx.accept(ConsumerEx.java:46) ~[hazelcast-jet-4.2-SNAPSHOT.jar:4.2-SNAPSHOT]
	at com.hazelcast.jet.impl.processor.TransactionPoolSnapshotUtility.restoreFromSnapshot(TransactionPoolSnapshotUtility.java:241) ~[hazelcast-jet-4.2-SNAPSHOT.jar:4.2-SNAPSHOT]
	at com.hazelcast.jet.impl.processor.TwoPhaseSnapshotCommitUtility.restoreFromSnapshot(TwoPhaseSnapshotCommitUtility.java:178) ~[hazelcast-jet-4.2-SNAPSHOT.jar:4.2-SNAPSHOT]
	at com.hazelcast.jet.kafka.impl.WriteKafkaP.restoreFromSnapshot(WriteKafkaP.java:171) ~[hazelcast-jet-kafka-4.2-SNAPSHOT.jar:?]
	at com.hazelcast.jet.impl.processor.ProcessorWrapper.restoreFromSnapshot(ProcessorWrapper.java:133) ~[hazelcast-jet-4.2-SNAPSHOT.jar:4.2-SNAPSHOT]
	at com.hazelcast.jet.impl.execution.ProcessorTasklet.processInbox(ProcessorTasklet.java:412) ~[hazelcast-jet-4.2-SNAPSHOT.jar:4.2-SNAPSHOT]
	at com.hazelcast.jet.impl.execution.ProcessorTasklet.stateMachineStep(ProcessorTasklet.java:289) ~[hazelcast-jet-4.2-SNAPSHOT.jar:4.2-SNAPSHOT]
	at com.hazelcast.jet.impl.execution.ProcessorTasklet.stateMachineStep(ProcessorTasklet.java:284) ~[hazelcast-jet-4.2-SNAPSHOT.jar:4.2-SNAPSHOT]
	at com.hazelcast.jet.impl.execution.ProcessorTasklet.stateMachineStep(ProcessorTasklet.java:266) ~[hazelcast-jet-4.2-SNAPSHOT.jar:4.2-SNAPSHOT]
	at com.hazelcast.jet.impl.execution.ProcessorTasklet.call(ProcessorTasklet.java:240) ~[hazelcast-jet-4.2-SNAPSHOT.jar:4.2-SNAPSHOT]
	at com.hazelcast.jet.impl.execution.TaskletExecutionService$BlockingWorker.run(TaskletExecutionService.java:293) ~[hazelcast-jet-4.2-SNAPSHOT.jar:4.2-SNAPSHOT]
	... 5 more
```

Steps to reproduce:
1) clone https://github.com/olukas/hazelcast-jet-ansible-tests/tree/kafka-version-reproducer and build it by `mvn clean install -DskipTests`
2) prepare Jet - take latest Jet 4.2-SNAPSHOT version and copy `hazelcast-jet-kafka` and `hazelcast-jet-cdc-debezium` jars from `opt` to `lib` to have them both on classpath and start two Jet servers by `./jet-start`
3) start kafka
4) submit job by `./jet submit -v $PATH_TO_CLONED_SOAK_TESTS_REPO/stateful-map-test/target/stateful-map-test-4.2-SNAPSHOT.jar` - it starts a test which takes 3 minutes
5) during test duration try to stop and start any of Jet servers, exception is thrown and job is failed. This does not happen it hazelcast-jet-cdc-debezium is removed from `lib` (when it is not on classpath)

--- a/extensions/cdc-debezium/pom.xml
+++ b/extensions/cdc-debezium/pom.xml
@@ -34,7 +34,6 @@
 
     <properties>
         <debezium.version>1.1.0.Final</debezium.version>
-        <kafka.connect.version>2.3.1</kafka.connect.version>
         <testcontainers.version>1.13.0</testcontainers.version>
         <log4j2.slf4j.binding.version>2.13.2</log4j2.slf4j.binding.version>
 
@@ -96,13 +95,13 @@
         <dependency>
             <groupId>org.apache.kafka</groupId>
             <artifactId>connect-api</artifactId>
-            <version>${kafka.connect.version}</version>
+            <version>${kafka.version}</version>
         </dependency>
 
         <dependency>
             <groupId>org.apache.kafka</groupId>
             <artifactId>connect-transforms</artifactId>
-            <version>${kafka.connect.version}</version>
+            <version>${kafka.version}</version>
         </dependency>
 
         <dependency>

